velox/connectors/hive/HiveConnectorUtil.cpp

```diff
@@ -628,6 +629,14 @@ bool testFilters(
           VELOX_CHECK(handlesIter != partitionKeysHandle.end());
 
           // This is a non-null partition key
+          if (handlesIter->second->dataType()->isDate()) {
```

Comment:
Can you add this logic to `applyPartitionFilter` instead?

Reply:
Thanks for your review. Updated as suggested.